### PR TITLE
Add Handler factories

### PR DESCRIPTION
**Pull request by lkostrowski**
This PR contributes to lean app templates, allowing sleek and minimal creation of handlers in Next

In the future, we will transform SDK to monorepo with SDK core + separate React, Next, Remix etc

### CODE REVIEW

File: docs/api-handlers.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Currently, Saleor heavily relies on Next.js, but in the future, other platforms will be supported.
+
+## Required handlers
+
+Saleor requires 2 endpoints to be available for a standalone app:
+
+- Manifest endpoint - Saleor asks for App information like name and permissions
+- Register endpoint - Saleor uses it to exchange authorization data
+
+TODO - any other endpoints?
```
Comment (lkostrowski) [inline docs/api-handlers.md:14]: @krzysztofwolski can you confirm only manifest and register are required by Saleor?

File: src/handlers/next/create-manifest-handler.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        };
+      },
+    });
+
+    const response = await handler(req, res);
+
+    /**
+     * TODO Fixme - why handler returns undefined?
+     */
+    expect(response).toEqual({});
```
Comment (lkostrowski) [inline src/handlers/next/create-manifest-handler.test.ts:36]: @zaiste can you help me why this test fails? Inside handler itself everything logs fine, but `Response.OK` seems to return undefined

File: tsup.config.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { defineConfig } from "tsup";
+
+export default defineConfig({
```
Comment (lkostrowski) [inline tsup.config.ts:3]: Also extracted config from package.json and specify entry points. With upcoming updates we can changed imports to root package for easier moving files around later

File: docs/api-handlers.md (hunk 1 above)
Comment (krzysztofwolski) [inline docs/api-handlers.md:14]: Also the [appUrl](https://docs.saleor.io/docs/3.x/developer/extending/apps/manifest) which is used to display a view in the dashboard when app is opened. Not 100% sure if Saleor allow to create app without it 🤔

File: docs/api-handlers.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+Saleor Apps are meant to work in serverless environment, where Cloud Functions are the foundations of server-side code.
+
+Currently, Saleor heavily relies on Next.js, but in the future, other platforms will be supported.
+
+## Required handlers
+
+Saleor requires 2 endpoints to be available for a standalone app:
+
+- Manifest endpoint - Saleor asks for App information like name and permissions
```
Comment (krzysztofwolski) [inline docs/api-handlers.md:11]:
```suggestion
- Manifest endpoint - Returns JSON object with app properties, like its name or permissions. [Read more](https://docs.saleor.io/docs/3.x/developer/extending/apps/manifest)
```

File: docs/api-handlers.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+- Register endpoint - Saleor uses it to exchange authorization data
```
Comment (krzysztofwolski) [inline docs/api-handlers.md:12]:
```suggestion
- Register endpoint - During the installation process, Saleor sends `POST` request with auth token to this endpoint. [Read more](https://docs.saleor.io/docs/3.x/developer/extending/apps/installing-apps#installation-using-graphql-api)
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: docs/api-handlers.md (hunk 1 above)
Comment (lkostrowski) [reply docs/api-handlers.md:14]: But appUrl is not endpoint, right? Its set by the factory itself

**Review by zaiste — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**